lib/meli-client.js

```diff
@@ -0,0 +1,111 @@
+const Promise = require('bluebird')
+const req = require('request-promise')
+const Account = require('../model/account')
+const {refresh} = require('../config/meli-auth')
+
+const GET_ORDERS_URL = `https://api.mercadolibre.com/orders/search`
+
+class MeliClient {
+  constructor() {
+    this.accounts = []
+  }
+
+  /**
+   *
+   * @param filters
+   * @returns {Promise<void>}
+   */
+  async getOrders({accounts = [], id} = {}) {
+    await this._updateAccounts()
+    const filterAccountsIds = accounts.map(a => a.id)
+    const filteredAccounts = filterAccountsIds.length > 0 ?
+      this.accounts.filter(acc => filterAccountsIds.includes(acc.id)) :
+      this.accounts
+
+    const authenticatedAccounts = await this.authenticate(filteredAccounts)
+
+    const reqOptionsBuilder = account => ({
+      uri: GET_ORDERS_URL,
+      qs: {
+        seller: account.id,
+        access_token: account.auth.accessToken,
+        sort: 'date_desc',
+        q: id
+      },
+      json: true
+    })
+
+    let orders = []
+    try {
+      orders = await Promise.map(authenticatedAccounts, async acc => ({
+        account: acc,
+        response: await req(reqOptionsBuilder(acc))
+          .catch(e => {
+            e.message = `Problem with meli orders request for ${acc.nickname}. ${e.message} `
+            throw e
+          })
+      }))
+    } catch (e) {
+      console.error(e)
+    }
+
+    return orders
+  }
+
+  addAccount(account) {
+    if (account === null || account === undefined || !account.nickname) {
```

Comment:
```js
 if (!account || !account.nickname) {
```
is simpler

Reply:
At first I preferred that path too, but I remember seeing a complaint somewhere... that when you need to check for null or undefined it's better to do it explicitly than relying on javascript falsy evaluation

Anyway I'm not finding this reference now and XO isn't complaining neither so I'll change it back to the "simpler" way